File: codility/PassingCars/PassingCars.py
# Author: Liang Lin
# Email: lianglin@outlook.com
# Date: 2016-04-01
class Solution:
    def solution(self, A):
        n = len(A)
        counter = 0
        eastcar = 0
        threshold = 1000000000
        for i in range(n):
            if A[i] == 0:
                eastcar += 1
            else:
                counter += eastcar
                if counter > threshold:
                     return -1
        return counter

sol = Solution()
print(sol.solution([0,1,0,1,1,0,1]))

File: codility/5.PrefixSums/PassingCars/PassingCars.java
/**
 * Author: Liang Lin
 * Email: lianglin@outlook.com
 * Date: 2016-04-01
 */
Class Solution {
    public int solution(int[] A) {
        int counter = 0;
        int eastcar = 0;
        int threshold = 1000000000;
        for (int i = 0; i < A.length; i++) {
            if(A[i] == 0) {
                eastcar ++;
            } else {
                counter += eastcar;
                if(counter > threshold) {
                    return -1;
                }
            }
        }
        return counter;
    }
}
